src/kaitai_spec/attribute/tests.rs

```diff
@@ -0,0 +1,191 @@
+#[cfg(test)]
+mod tests {
+    use super::*;
+#[test]
+    fn simple_de() {
+        let yaml = "
+            id: magic1
+            doc: example
+            doc-ref: ref1
+            contents: [0xca, 0xfe, 0xba, 0xbe]
+        ";
+
+        let attribute: Attribute = serde_yaml::from_str(yaml).unwrap();
+
+        let Attribute {
+            id,
+            doc,
+            doc_ref,
+            type_,
+        } = attribute;
+
+        assert_eq!(id, "magic1");
+        assert_eq!(doc, Some("example".to_string()));
+        assert_eq!(doc_ref, Some("ref1".to_string()));
+        match type_ {
+            AttributeType::Contents(contents) => {
+                assert_eq!(contents, vec![0xCA, 0xFE, 0xBA, 0xBE]);
+            }
+        }
+    }
+#[test]
+    fn without_id() {
+        let yaml = "
+            doc: example
+            doc-ref: ref1
+            contents: JFIF
+        ";
+        let attribute: Attribute = serde_yaml::from_str(yaml).unwrap();
+        let Attribute {
+            id,
+            doc,
+            doc_ref,
+            type_,
+        } = attribute;
+
+        assert_eq!(id, "");
+        assert_eq!(doc, Some("example".to_string()));
+        assert_eq!(doc_ref, Some("ref1".to_string()));
+        match type_ {
+            AttributeType::Contents(contents) => {
+                assert_eq!(contents, b"JFIF".to_vec());
+            }
+        }
+    }
+#[test]
+    fn string_case() {
+        let yaml = "
+        id: magic1
+        doc: example
+        doc-ref: ref1
+        contents: JFIF
+    ";
+
+    let attribute: Attribute = serde_yaml::from_str(yaml).unwrap();
+
+    let Attribute {
+        id,
+        doc,
+        doc_ref,
+        type_,
+    } = attribute;
+
+    assert_eq!(id, "magic1");
+    assert_eq!(doc, Some("example".to_string()));
+    assert_eq!(doc_ref, Some("ref1".to_string()));
+    match type_ {
+        AttributeType::Contents(contents) => {
+            assert_eq!(contents, b"JFIF".to_vec());
+        }
+    }
+    }
+#[test]
+    fn extra_field() {
+        let yaml = "
+        id: magic1
+        doc: example
+        doc-ref: ref1
+        contents: JFIF
+        extra_field:
+    ";
+
+    let result: Result<Attribute, _> = serde_yaml::from_str(yaml);
+
+    assert!(!result.is_ok());
+}
+#[test]
+    fn string_and_number_case() {
+        let yaml = "
+        id: magic1
+        doc: example
+        doc-ref: ref1
+        contents: [CAFE, 0, BABE]
+    ";
+
+    let attribute: Attribute = serde_yaml::from_str(yaml).unwrap();
+
+    let Attribute {
+        id,
+        doc,
+        doc_ref,
+        type_,
+    } = attribute;
+
+    assert_eq!(id, "magic1");
+    assert_eq!(doc, Some("example".to_string()));
+    assert_eq!(doc_ref, Some("ref1".to_string()));
+    match type_ {
+        AttributeType::Contents(contents) => {
+            assert_eq!(contents, vec![67, 65, 70, 69, 0, 66, 65, 66, 69]);
+        }
+    }
+    }
+#[test]
```

Comment:
and please separate test functions with empty lines to increase code readability